Realm/RLMResults.mm

```diff
@@ -427,3 +428,25 @@ - (BOOL)isAttached
 
 @implementation RLMLinkingObjects
 @end
+
+@interface RLMResults (Handover) <RLMThreadConfined_Private>
+@end
+
+@implementation RLMResults (Handover)
+
+- (realm::AnyThreadConfined)rlm_handoverData {
+    return translateErrors([&] { return AnyThreadConfined(_results); }); // TODO: Do we need more translate errors thing?
+}
+
+- (NSString *)rlm_handoverMetadata {
+    return self.objectSchema.className;
```

Comment:
Why is passing this extra data necessary? `Results::get_object_type` on the handed-over instance should give you the same information.